src/main/java/picard/sam/markduplicates/EstimateLibraryComplexity.java

```diff
@@ -210,26 +207,31 @@ public static int getSizeInBytes() {
             return new PairedReadCodec();
         }
 
-        void initHashes(int maxReadLength, int hashLength, int skippedBases) {
-            dupCandidates = new HashSet<>();
-            hashes1 = getHashes(read1, maxReadLength, hashLength, skippedBases);
-            hashes2 = getHashes(read2, maxReadLength, hashLength, skippedBases);
+        void initHashes(int numberOfHashes, int skippedBases, int minReadLength) {
+            hashes1 = getHashes(read1, numberOfHashes, skippedBases, minReadLength);
+            hashes2 = getHashes(read2, numberOfHashes, skippedBases, minReadLength);
         }
 
-        // Split read by (MAX_DIFF_RATE * read.length + 1) parts and hash each part
-        private int[] getHashes(byte[] read, int maxReadLength, int hashLength, int skippedBases) {
-            int maxLengthWithoutHead = maxReadLength - skippedBases;
-            int hashNum = (Math.min(read.length - skippedBases, maxLengthWithoutHead)) /
-                    hashLength;
-            int[] hashValues = new int[hashNum];
-            for (int i = 0; i < hashNum; ++i) {
-                int st = skippedBases + i * hashLength;
-                int end = st + hashLength;
-
-                // use custom hash to avoid using System.arraycopy
+        // Split read by numberOfHashes parts and hash each part
+        // For instance:
+        //        0 1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 19
+        // read = A C G T A C G T A C G  T  A  C  G  T  A  C  G  T
+        // numberOfHashes = 5
+        // skippedBases = 1
+        // minReadLength = 15
+        // So, method returns hashValues with 5 hash value
```

Comment:
unless I am misunderstanding something, the example in the comment does not make use of skippedBases=1